### ISSUE TITLE
Chrome extension crashing

### ISSUE THREAD

**Issue by sajan45** [SchizoDuckie/DuckieTV#905]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
Chrome Extension (Tested both New tab and Browser action)
Chrome version "59.0.3071.86 (Official Build) (64-bit)"

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
1.1.4

**What is your Operating System (Windows, Mac, Linux, Android)**
Linux (Ubuntu 16.04)

**Describe the problem you are having and steps to reproduce if available**
Maybe it updated in background. So I was trying to open it (browser action mode) today after a week of last use. It just opened and closed(like crashing) then disappear from Chrome tool bar. When I checked in Chrome extensions menu, it showed "That this extension may have been corrupted" below the name. Tried reinstalling the same version and again tried the new tab version but same problem, crashed and closed and showed that its corrupted.
Now installed standalone version for Linux, its working fine.

**Attach any DuckieTV statistics or Developer Console logs if available**
Its closing immediately so cant get any console log.

**Comment by yasserkaddour** [comment]
Same problem here since the update to chrome 59.

**Comment by garfield69** [comment]
Curious.     
Unfortunately I don't have an x64 computer, so I cannot replicate your issue. On my Ubuntu 16.10 x32, I use Chromium (since Google Chrome does not do x32 releases anymore). Not problems with using the DuckieTV 1.1.4 extension on Chromium 58 so far. Can't find a Chromium 59 stable release as yet.      

So are you able to install a DuckieTV Nightly Chrome Extension ? https://github.com/SchizoDuckie/DuckieTV/wiki/How-to-Install-a-Development-version-of-the-DuckieTV-Chrome-Browser-Extension-NIGHTLY

**Comment by yasserkaddour** [comment]
Hi @garfield69 I tried the nightly build but I still have the same problem, installing by drag and dropping the build files in the extensions page  work, but the extension disappear every time I restart chrome. 
Thanks for helping.

**Comment by sajan45** [comment]
@garfield69 I tried the nightly build too. Same problem as before and as stated by YasserKaddour.

**Comment by garfield69** [comment]
Ah. You can only install extensions from the chrome web store. So have you tried clicking on this link https://chrome.google.com/webstore/detail/duckietv-browser-action-m/ikhklcamhmfmjbhckdjoabiepglbhaid, and then clicking on the *add-to-chrome* button?
![image](https://user-images.githubusercontent.com/6933240/27052440-ced29e2e-500d-11e7-8d99-2aea3b48a4b6.png)

**Comment by garfield69** [comment]
# addendum: 
Signs are starting to show up on the web that many other people are discovering chrome extension issues when their chrome updated from 58 to 59. examples: https://community.norton.com/en/forums/extension-may-have-been-corrupted-error-message-toolbar-extension-after-updating-chrome-v59 and https://github.com/dasilva333/TowerGhostForDestiny/issues/704    
It's not clear yet what the solution is, or why some are affected and not others.     
I'll keep an eye out over the next couple days and see if a consensus appears.     
In the mean time you can at least continue to use DuckieTV Standalone to get your TV fix ;-)

**Comment by sajan45** [comment]
@garfield69 yes, I followed above steps and installed nightly build from chrome web store .As you mentioned, maybe it is something to do with Chrome 59. For now I will continue with standalone version.

**Comment by garfield69** [comment]
found https://bugs.chromium.org/p/chromium/issues/detail?id=720597 which seems the likely cause of the problem.     
I am not sure when chrome will push this fix out, but I assume it will be in the next update after 59.0.3071.86

**Comment by yasserkaddour** [comment]
Thanks for investigating,
the culprit is `templates/todopicker.html` with his 4096 byte, #907 will fix the nightly build at least.

**Comment by SchizoDuckie** [comment]
Ffs talk about crazy bugs. Nice work y'all

**Comment by Js41637** [comment]
Are you saying it was crashing because it was exactly 4096k large, what are the odds

**Comment by yasserkaddour** [comment]
Yep, odd bug :bug:  indeed. 
Summary from the [chrome issue](https://bugs.chromium.org/p/chromium/issues/detail?id=720597)

> files that are of size equals multiple of content hash's block_size (4096 bytes) always fail content verification due to bug.

Anyway I confirmed that the issue is fixed locally, and I think we can close this.

Thanks

**Comment by garfield69** [comment]
@sajan45  @YasserKaddour  could either of you please confirm that the latest Nightly from the google store is no longer rejected by your chrome browser extension's verification check? Thanks.

**Comment by sajan45** [comment]
@garfield69 I installed the nightly build version 3017.6.16.91 but it still has the same issue.
![duckie](https://user-images.githubusercontent.com/10629412/27221015-b3f7db70-52a4-11e7-9c3c-35f08a365947.png)

**Comment by yasserkaddour** [comment]
Same here, the nightly build is still buggy, I thought my patch fixed the problem, but after further investigation, what fixed the problem was using a decompressed folder instead of the zip build.

I tried `google-chrome --extension-content-verification=encorce --enable-logging=stderr` this is what I got:

> [5656:5656:0616/124120.156008:INFO:CONSOLE(18)] "This is a first install!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/background.js (18)
[5656:5656:0616/124120.157292:INFO:CONSOLE(7)] "This is a first install!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/launch.js (7)
[5656:5656:0616/124133.325955:INFO:CONSOLE(91)] "%cDuckieTV", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/deps.js (91)
[5656:5656:0616/124133.543041:INFO:CONSOLE(8206)] "Registering torrent client: Deluge", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.544140:INFO:CONSOLE(8206)] "Registering torrent client: qBittorrent (pre3.2)", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.545231:INFO:CONSOLE(8206)] "Registering torrent client: qBittorrent 3.2+", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.546258:INFO:CONSOLE(8206)] "Registering torrent client: Ktorrent", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.548101:INFO:CONSOLE(8206)] "Registering torrent client: rTorrent", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.549165:INFO:CONSOLE(8206)] "Registering torrent client: Tixati", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.550100:INFO:CONSOLE(8206)] "Registering torrent client: Transmission", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.553115:INFO:CONSOLE(8206)] "Registering torrent client: uTorrent Web UI", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.553795:INFO:CONSOLE(8206)] "Registering torrent client: Vuze", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (8206)
[5656:5656:0616/124133.565008:INFO:CONSOLE(11043)] "SettingsService.changeLanguage", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (11043)
[5656:5656:0616/124133.570060:INFO:CONSOLE(10487)] "Updated localStorage with SNR name and date exceptions.", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10487)
[5656:5656:0616/124133.570568:INFO:CONSOLE(10574)] "Initializing Xross Entity Mapping (http://thexem.de/) service for Scene Name episode format.", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10574)
[5656:5656:0616/124133.572331:INFO:CONSOLE(10595)] "Executing the 1.1 migration to populate watched and notWatchedCount entities", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10595)
[5656:5656:0616/124133.572737:INFO:CONSOLE(10613)] "Executing the 1.1.4cleanupOrphanedSeasons to remove orphaned seasons", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10613)
[5656:5656:0616/124133.573102:INFO:CONSOLE(10631)] "Executing the 1.1.4cleanupOrphanedEpisodes to remove orphaned episodes", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10631)
[5656:5656:0616/124133.573543:INFO:CONSOLE(10656)] "Executing the 1.1.4refresh to update Trakt_id for all Series, Seasons and Episodes", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10656)
[5656:5656:0616/124133.574057:INFO:CONSOLE(10662)] "Executing 1.1.4qBittorrentPre32 to rename qBittorrent to qBittorrent (pre3.2)", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10662)
[5656:5656:0616/124133.574563:INFO:CONSOLE(10668)] "1.1.4qBittorrentPre32 done!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10668)
[5656:5656:0616/124133.574983:INFO:CONSOLE(10674)] "Executing 1.1.4refreshTrendingCache to refresh the TraktTV Trending Cache with trakt_id", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10674)
[5656:5656:0616/124133.575356:INFO:CONSOLE(10678)] "1.1.4refreshTrendingCache done!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10678)
[5656:5656:0616/124133.575739:INFO:CONSOLE(10684)] "Executing 1.1.4updateDelugeAuth to set deluge.use_auth to true", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10684)
[5656:5656:0616/124133.576324:INFO:CONSOLE(10687)] "1.1.4updateDelugeAuth done!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10687)
[5656:5656:0616/124133.576696:INFO:CONSOLE(10693)] "Executing 1.1.4updateTransmissionPath to clone transmission.key to transmission.path", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10693)
[5656:5656:0616/124133.577400:INFO:CONSOLE(10705)] "1.1.4updateTransmissionPath done!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10705)
[5656:5656:0616/124133.577821:INFO:CONSOLE(10726)] "Executing 1.1.4TorrentHashListCleanup to remove obsolete torrentHashes from TorrentHashListService", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (10726)
[5656:5656:0616/124133.579497:INFO:CONSOLE(11431)] "Enabling torrent auto-stop!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (11431)
[5656:5656:0616/124133.580768:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.581355:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.581922:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.582504:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.583064:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.583621:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.584182:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.584722:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.585482:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.586203:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.586695:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.587197:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.587721:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.588259:INFO:CONSOLE(15295)] "Registering torrent search engine:", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (15295)
[5656:5656:0616/124133.590095:INFO:CONSOLE(11173)] "Register new storage sync target!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (11173)
[5656:5656:0616/124133.590749:INFO:CONSOLE(17777)] "TraktTV storage sync target initialized!", source: chrome-extension://cdfkaloficjmdjbgmckaddgfcghgidei/dist/app.js (17777)
[5656:5656:0616/124133.663161:ERROR:CONSOLE(1)] "Uncaught ReferenceError: uber_frame is not defined", source:  (1)

Tell me if there is anything else I can do to help the debugging.

**Comment by garfield69** [comment]
@YasserKaddour  was `I tried google-chrome --extension-content-verification=encorce --enable-logging=stderr` a typo or did you really use `encorce` instead of `enforce`?

**Comment by garfield69** [comment]
Your log is interesting. It shows that your DuckieTV was running, but fell over when something tried referencing `uber_frame`, which is not something DuckieTV source references directly, so it must be an internal chrome reference... I'll do some poking around and see what I can find.

**Comment by garfield69** [comment]
could you paste the following in your chrome address box `chrome://flags/#extension-content-verification` and confirm that your setting is `Default` ?

**Comment by garfield69** [comment]
@sajan45  Did you try trashing the Nightly extensions, shutting down chrome, starting it up again (which will clear the extension data space), and installing the nightly afresh?

**Comment by sajan45** [comment]
@garfield69 Did that that again. Trashed, closed chrome and started again, installed nightly build . Issue still exists.

**Comment by yasserkaddour** [comment]
`extension-content-verification` is set to `Default`, and yes I did all that.
I don't think it's related to to the `uber_frame` error because there are times when I don't get this error, and my understanding is it wouldn't trigger an `this extension may have been corrupted` message.

I will try to further investigate later this week.

**Comment by lothas** [comment]
I started getting the same problem a few days ago.
I'm running Version 60.0.3112.90 (Official Build) (64-bit) on Ubuntu 16.04.
DuckieTV - 'Browser Action' mode 1.1.4
This extension may have been corrupted.

**Comment by garfield69** [comment]
@lothas  Yeah, seems to be a problem that plagues the linux chrome browser.      
Until this is resolved, you may wish to use the Standalone version of DuckieTV instead: https://github.com/SchizoDuckie/DuckieTV/releases/download/1.1.4/DuckieTV-1.1.4-ubuntu-x64.deb       
If you previously took a DuckieTV *backup*, then you can *import* it on your Standalone which will restore your series, episodes and settings.      
If you do not have a backup and don't want to start from scratch, then let me know and I'll knock up some instructions for you to try migrating the episode data across. You will still need to update the settings by hand.

**Comment by lothas** [comment]
Hi and thanks for your reply. And by the way, thanks for DuckieTV!
I tried installing the standalone version from your link but I get an error. With the gui, it just doesn't install. Using dpkg I get this:
/usr/share/menu/DuckieTV: 1: /usr/share/menu/DuckieTV: Syntax error: word unexpected (expecting ")")

**Comment by garfield69** [comment]
Oh yeah, forgot the 1.1.4 debian build had installer script errors..... how about https://github.com/DuckieTV/Nightlies/releases/download/nightly-201708090130/DuckieTV-201708090130-ubuntu-x64.deb   ?
I dont have a x64 linux, but I have successfully installed the x32 debian on my ubuntu 17.04 sandbox. Let me know how it goes ;-)

**Comment by lothas** [comment]
Same error as before, right after:
Processing triggers for menu (2.1.47ubuntu1) ...
/usr/share/menu/DuckieTV: 1: /usr/share/menu/DuckieTV: Syntax error: word unexpected (expecting ")")

**Comment by garfield69** [comment]
Well that is a puzzle. Our buildbot uses the same debian menu file for both the x32 and x64 builds.     
the Debian Menu System syntax as described here https://www.debian.org/doc/packaging-manuals/menu.html/ch3.html is what we are using.     
Our menu file contains:
```
?package(DuckieTV): needs="X11" \
  section="Applications/Network" \
  title="DuckieTV" \
  command="/opt/DuckieTV/bin/DuckieTV" \
  icon="/usr/share/pixmaps/DuckieTV.xpm"
```
which as far as I can tell conforms to the expected syntax.  Review welcome.        
When I run the installer from a terminal on my x32 ubuntu this is what I get:
```
garfield69@PC2-Ubuntu:~/Downloads$ sudo dpkg -i DuckieTV-201708100130-ubuntu-x32.deb
[sudo] password for garfield69: 
(Reading database ... 264108 files and directories currently installed.)
Preparing to unpack DuckieTV-201708100130-ubuntu-x32.deb ...
Unpacking duckietv:x32 (201708100130.nightly) over (201708060130.nightly) ...
Setting up duckietv:x32 (201708100130.nightly) ...
Processing triggers for bamfdaemon (0.5.3+17.04.20170406-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for gnome-menus (3.13.3-6ubuntu5) ...
Processing triggers for desktop-file-utils (0.23-1ubuntu2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
garfield69@PC2-Ubuntu:~/Downloads$
```
could you show me what you get when you install like I did from a terminal?
the only thing I can think of is that maybe you need to check for updates for your dpkg product?

**Comment by lothas** [comment]
Thanks. It seems like there's some problem with my home PC. I tried installing it in another PC (also with 64-bit Ubuntu) and it worked. I'll check later if I can get it to install at home.

**Comment by lothas** [comment]
Quick update: It seems one of my install attempts worked because DuckieTV showed up in Synaptic as installed :+1:

**Comment by garfield69** [comment]
regarding the `This extension may have been corrupted` issue on linux and macs:
### news:
  [720597](https://bugs.chromium.org/p/chromium/issues/detail?id=720597)  (the %4096 bug) is fixed in M61 (the fix landed in 61.0.3124.0) according to [736699](https://bugs.chromium.org/p/chromium/issues/detail?id=736699) and [754282](https://bugs.chromium.org/p/chromium/issues/detail?id=754282).  \o/ 
### additional related ticket to 720597
[736699](https://bugs.chromium.org/p/chromium/issues/detail?id=736699) which has the following:         
>Reason 3 indicates a hash mismatch, but I think it's more likely that the issue is due to "search-csm.png" being requested but "Search-csm.png" being present in the archive      

This affects chrome extensions on linux and mac in particular. The ticket also explains that the hash checking applies only to the web-store, locally loaded extensions (in developer mode) do not get checked.

### to do
- [x] check that all extension images and icons filenames are loaded with the correct case-sensitive names.
